File: java-service/src/main/java/com/analyticore/domain/repository/JobRepository.java
package com.analyticore.domain.repository;

import com.analyticore.domain.model.AnalysisResult;
import com.analyticore.domain.model.Job;

import java.util.Optional;

public interface JobRepository {
    Optional<Job> findById(String jobId);
    void updateStatus(String jobId, String status);
    void saveResults(String jobId, AnalysisResult result);
}

File: python-service/app/domain/repositories/job_repository.py
from abc import ABC, abstractmethod
from typing import Optional

from app.domain.entities.job import Job


class JobRepository(ABC):
    @abstractmethod
    def create(self, text: str) -> Job:
        pass

    @abstractmethod
    def find_by_id(self, job_id: str) -> Optional[Job]:
        pass
